install.sh

```diff
@@ -0,0 +1,322 @@
+#!/bin/sh
+
+# container-use installer script
+# Downloads and installs the appropriate binary for your system
+
+set -e
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Configuration
+REPO="dagger/container-use"
+BINARY_NAME="cu"
+INSTALL_DIR=""
+
+# Helper functions
+log_info() {
+    printf "${BLUE}ℹ️  %s${NC}\n" "$1"
+}
+
+log_success() {
+    printf "${GREEN}✅ %s${NC}\n" "$1"
+}
+
+log_warning() {
+    printf "${YELLOW}⚠️  %s${NC}\n" "$1"
+}
+
+log_error() {
+    printf "${RED}❌ %s${NC}\n" "$1"
+}
+
+# Check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check dependencies
+check_dependencies() {
+    log_info "Checking dependencies..."
+
+    if ! command_exists docker; then
+        log_error "Docker is required but not installed."
+        log_info "Please install Docker from: https://docs.docker.com/get-started/get-docker/"
+        exit 1
+    fi
+
+    log_success "Docker found"
+}
+
+# Detect operating system
+detect_os() {
+    local os
+    case "$(uname -s)" in
+        Linux*)     os="linux";;
+        Darwin*)    os="darwin";;
+        CYGWIN*|MINGW32*|MSYS*|MINGW*)
+            log_error "Windows is not supported"
+            log_info "container-use uses Unix syscalls and requires Linux or macOS"
+            exit 1;;
+        *)
+            log_error "Unsupported operating system: $(uname -s)"
+            exit 1;;
+    esac
+    echo "$os"
+}
+
+# Detect architecture
+detect_arch() {
+    local arch
+    case "$(uname -m)" in
+        x86_64|amd64)   arch="amd64";;
+        arm64|aarch64)  arch="arm64";;
+        *)
+            log_error "Unsupported architecture: $(uname -m)"
+            exit 1;;
+    esac
+    echo "$arch"
+}
+
+# Check for existing cu command and warn about conflicts
+check_existing_cu() {
+    local found_binary=$(command -v "$BINARY_NAME" 2>/dev/null || echo "")
+
+    if [ -n "$found_binary" ]; then
+        # Only warn about system paths (user paths could be previous container-use installations)
+        case "$found_binary" in
+            /usr/bin/* | /bin/* | /usr/local/bin/*)
+                log_warning "Existing 'cu' command found at $found_binary"
+                log_warning "This appears to be a system 'cu' command (likely Taylor UUCP)"
+                log_warning "After installation, you may need to run 'hash -r' to clear command cache"
+                log_info "Or use the full path: \$HOME/.local/bin/cu"
+                ;;
+        esac
+
+        log_info "Installation will continue..."
+        echo ""
+    fi
+}
+
+# Find the best installation directory
+find_install_dir() {
+    # Prefer ~/.local/bin if it's in PATH
+    if echo "$PATH" | grep -q "$HOME/.local/bin"; then
+        echo "$HOME/.local/bin"
+        return
+    fi
+
+    # Look for writable directories in PATH
+    OLD_IFS="$IFS"
+    IFS=':'
+    for dir in $PATH; do
```

Comment:
This was the old logic, see hack/install.sh for new logic.

Also if you no longer use hack/ scripts, make sure you remove them.